### PR TITLE
feat: Pottery

### PR DESCRIPTION
**Pull request by ChefMomota**

### CODE REVIEW

File: src/views/Pottery/components/HowToPlay.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  ]
+
+  return (
+    <HowToPlayContainer>
+      <Text fontSize="40px" mb="24px" color="secondary" bold>
+        {t('How to Play')}
+      </Text>
+      <Text textAlign="center">
+        {t(
+          'Deposite CAKE to get your Pottery tickets. More CAKE deposited, higher the chance of winning (and of course, higher the rewards)!',
```
Comment (memoyil) [inline src/views/Pottery/components/HowToPlay.tsx:79]: Typo

File: src/views/Pottery/components/Banner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  const cakePriceBusd = usePriceCakeBusd()
+  const { publicData } = usePotteryData()
+  const { getLockedApy } = useVaultApy()
+
+  const prizeInBusd = publicData.totalPrize.times(cakePriceBusd)
+  const prizeTotal = getBalanceNumber(prizeInBusd)
+
+  const apyDisplay = useMemo(() => {
+    const apy = getLockedApy(weeksToSeconds(10))
+    return `${Number(apy).toFixed(2)}%`
```
Comment (chef-jojo) [inline src/views/Pottery/components/Banner/index.tsx:78]: avoid NAN

File: src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+  useEffect(() => {
+    setFinishedRoundInfoFetched(true)
+
+    timer.current = setInterval(() => {
+      if (selectedRoundId) {
+        dispatch(fetchPotteryRoundData(selectedRoundId))
+      }
+      clearInterval(timer.current)
+    }, 1000)
```
Comment (chef-jojo) [inline src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx:59]: could you tell me why do we need setInterval here?

Comment (ChefMomota) [reply src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx:59]: If not, we well cause every time user type will keep fetch data.

Comment (chef-jojo) [reply src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx:59]: useDebouncedChangeHandler can fulfill the goal?

Comment (ChefMomota) [reply src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx:59]: I think so, let me take a look.

File: .env.development [hunk 4]
```diff
@@ -17,6 +17,7 @@ NEXT_PUBLIC_GRAPH_API_PREDICTION = "https://api.thegraph.com/subgraphs/name/panc
 NEXT_PUBLIC_GRAPH_API_PREDICTION_CAKE = "https://api.thegraph.com/subgraphs/name/pancakeswap/prediction-cake"
 NEXT_PUBLIC_GRAPH_API_LOTTERY = "https://api.thegraph.com/subgraphs/name/pancakeswap/lottery"
 NEXT_PUBLIC_GRAPH_API_NFT_MARKET = "https://api.thegraph.com/subgraphs/name/pancakeswap/nft-market"
+NEXT_PUBLIC_GRAPH_API_POTTERY = "https://api.thegraph.com/subgraphs/name/chef-huan/pottery"
```
Comment (ChefMomota) [inline .env.development:20]: Revert before merge.
`NEXT_PUBLIC_GRAPH_API_POTTERY = "https://api.thegraph.com/subgraphs/name/pancakeswap/pottery"
`

File: src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx (hunk 3 above)
Comment (ChefMomota) [reply src/views/Pottery/components/FinishedRounds/AllHistoryCard/index.tsx]: @chef-jojo  Updated.

File: .env.production [hunk 5]
```diff
@@ -18,6 +18,7 @@ NEXT_PUBLIC_GRAPH_API_PREDICTION = "https://api.thegraph.com/subgraphs/name/panc
 NEXT_PUBLIC_GRAPH_API_PREDICTION_CAKE = "https://api.thegraph.com/subgraphs/name/pancakeswap/prediction-cake"
 NEXT_PUBLIC_GRAPH_API_LOTTERY = "https://api.thegraph.com/subgraphs/name/pancakeswap/lottery"
 NEXT_PUBLIC_GRAPH_API_NFT_MARKET = "https://api.thegraph.com/subgraphs/name/pancakeswap/nft-market"
+NEXT_PUBLIC_GRAPH_API_POTTERY = "https://api.thegraph.com/subgraphs/name/chef-huan/pottery"
```
Comment (ChefMomota) [inline .env.production:21]: Revert before merge.
NEXT_PUBLIC_GRAPH_API_POTTERY = "https://api.thegraph.com/subgraphs/name/pancakeswap/pottery"

File: src/hooks/useContract.ts [hunk 6]
```diff
@@ -371,3 +373,13 @@ export function usePairContract(pairAddress?: string, withSignerIfPossible?: boo
 export function useMulticallContract() {
   return useContract<Multicall>(getMulticallAddress(), multiCallAbi, false)
 }
+
+export const usePotterytValutContract = () => {
```
Comment (memoyil) [inline src/hooks/useContract.ts:377]: typo

File: src/state/pottery/fetchPotteryRound.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { request, gql } from 'graphql-request'
+import { GRAPH_API_POTTERY } from 'config/constants/endpoints'
+
+export const fetchPotteryFinisedRound = async (potteryRoundId: string) => {
```
Comment (memoyil) [inline src/state/pottery/fetchPotteryRound.ts:4]: typo

File: src/views/Pottery/index.tsx [hunk 8]
```diff
@@ -22,7 +23,13 @@ const Pottery: React.FC = () => {
       <HowToPlay />
       <PrizeFunds />
       <FAQ />
-      {createPortal(<ScrollToTopButton />, document.body)}
+      {createPortal(
+        <>
+          <ScrollToTopButton />
+          <SubgraphHealthIndicator subgraphName="chef-huan/pottery" />
```
Comment (ChefMomota) [inline src/views/Pottery/index.tsx:29]: Revert before merge.
subgraphName="pancakeswap/pottery"

File: src/views/Pottery/components/FAQ.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+            <Text fontSize="14px" color="textSubtle">
+              {t(
+                "All CAKE deposits will be locked for the same duration, the odds of your address winning the prize pool is simply proportional to your CAKE deposit over the total CAKE deposit of the whole Pottery. Each week, eight (8) addresses are drawn randomly based on their odds. Simply put, if user A deposited 1 CAKE, user B deposited 2 CAKE, then user B's odds of winning is twice of user A's.",
+              )}
+            </Text>
+          </FoldableText>
+          <FoldableText title={t('Is there any risk associated with depositing and playing Pottery?')} mt="24px">
+            <Text fontSize="14px" color="textSubtle">
+              {t('Is there any risk associated with depositing and playing Pottery?')}
+            </Text>
```
Comment (woodywu810) [inline src/views/Pottery/components/FAQ.tsx:71]: It repeated with the title.

image

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**